source/parser.hera

```diff
@@ -7575,6 +7575,48 @@ Init
       }
     }
 
+    function aggregateDuplicateBindings(bindings) {
+      const props = gatherRecursiveAll(bindings, n => n.type === "BindingMatchProperty")
+      const declarations = []
+
+      const propsGroupedByName = props.reduce((acc, p) => {
+        const { name } = p
+
+        const key = name.name
+        if (acc.has(key)) {
+          acc.get(key).push(p)
+        } else {
+          acc.set(key, [p])
+        }
+
+        return acc
+      }, new Map())
```

Comment:
My personal preference for writing this would be to have a `const acc = new Map` outside and use `for(...of...)` or `forEach` to iterate through `props`. But `reduce` is fine too; just slightly less efficient (?) and I find it a bit harder to read.